src/importlinter/domain/imports.py

```diff
@@ -27,6 +27,9 @@ def __init__(self, name: str) -> None:
         """
         self.name = name
 
+    def has_wildcard_expression(self) -> bool:
```

Comment:
My concern with this approach is that we're conflating modules with module expressions. I think it's important to keep this distinction, as we have done with the difference between the `DirectImport` and `ImportExpression` types.

With that in mind, I think we should add a new type called a `ModuleExpression` together with a `ModuleExpressionField`. We should then adjust the independence contract module field to be like so:

```
class IndependenceContract(Contract):
    modules = fields.ListField(subfield=fields.ModuleExpressionField())
    ...
```

The `IndependenceContract.check` method can then call a function in `contract_utils` (maybe called `resolve_module_expressions`?) which would turn an iterable of `ModuleExpression` objects into a set of `Module` objects. What do you think of that approach?